examples/window-app/telink/src/WindowCovering.cpp

```diff
@@ -0,0 +1,315 @@
+
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+using namespace chip::app::Clusters::WindowCovering;
+
+static const struct pwm_dt_spec sLiftPwmDevice = LIGHTING_PWM_SPEC_RGB_BLUE;
+static const struct pwm_dt_spec sTiltPwmDevice = LIGHTING_PWM_SPEC_RGB_GREEN;
+
+static constexpr uint32_t sMoveTimeoutMs{ 200 };
+constexpr uint8_t kDefaultMinLevel = 0;
+constexpr uint8_t kDefaultMaxLevel = 254;
+
+WindowCovering::WindowCovering()
+{
+    if (mLiftIndicator.Init(&sLiftPwmDevice, kDefaultMinLevel, kDefaultMaxLevel, kDefaultMinLevel) != CHIP_NO_ERROR)
+    {
+        LOG_ERR("Cannot initialize the lift indicator");
+    }
+    if (mTiltIndicator.Init(&sTiltPwmDevice, kDefaultMinLevel, kDefaultMaxLevel, kDefaultMinLevel) != CHIP_NO_ERROR)
+    {
+        LOG_ERR("Cannot initialize the tilt indicator");
+    }
+    mLiftIndicator.InitiateAction(PWMDevice::ON_ACTION, 0, nullptr);
+    mTiltIndicator.InitiateAction(PWMDevice::ON_ACTION, 0, nullptr);
+}
+
+void WindowCovering::DriveCurrentLiftPosition(intptr_t)
+{
+    NPercent100ths current{};
+    NPercent100ths target{};
+    NPercent100ths positionToSet{};
+
+    VerifyOrReturn(Attributes::CurrentPositionLiftPercent100ths::Get(Endpoint(), current) == EMBER_ZCL_STATUS_SUCCESS);
+    VerifyOrReturn(Attributes::TargetPositionLiftPercent100ths::Get(Endpoint(), target) == EMBER_ZCL_STATUS_SUCCESS);
+
+    UpdateOperationalStatus(MoveType::LIFT, ComputeOperationalState(target, current));
+
+    positionToSet.SetNonNull(CalculateSingleStep(MoveType::LIFT));
+    LiftPositionSet(Endpoint(), positionToSet);
+
+    // assume single move completed
+    Instance().mInLiftMove = false;
+
+    VerifyOrReturn(Attributes::CurrentPositionLiftPercent100ths::Get(Endpoint(), current) == EMBER_ZCL_STATUS_SUCCESS);
+
+    if (!TargetCompleted(MoveType::LIFT, current, target))
+    {
+        // continue to move
+        StartTimer(MoveType::LIFT, sMoveTimeoutMs);
+    }
+    else
+    {
+        // the OperationalStatus should indicate no-lift-movement after the target is completed
+        UpdateOperationalStatus(MoveType::LIFT, ComputeOperationalState(target, current));
+    }
+}
+
+chip::Percent100ths WindowCovering::CalculateSingleStep(MoveType aMoveType)
+{
+    EmberAfStatus status{};
+    chip::Percent100ths percent100ths{};
+    NPercent100ths current{};
+    OperationalState opState{};
+
+    if (aMoveType == MoveType::LIFT)
+    {
+        status  = Attributes::CurrentPositionLiftPercent100ths::Get(Endpoint(), current);
+        opState = OperationalStateGet(Endpoint(), OperationalStatus::kLift);
+    }
+    else if (aMoveType == MoveType::TILT)
+    {
+        status  = Attributes::CurrentPositionTiltPercent100ths::Get(Endpoint(), current);
+        opState = OperationalStateGet(Endpoint(), OperationalStatus::kTilt);
+    }
+
+    if ((status == EMBER_ZCL_STATUS_SUCCESS) && !current.IsNull())
+    {
+        static constexpr auto sPercentDelta{ WC_PERCENT100THS_MAX_CLOSED / 20 };
```

Comment:
this sound like a config about the number of steps 
this magic should not be here